File: Objective-C Version/InteractiveCardDemo/InteractiveCardDemo/InteractiveCardDemo/ViewController.m
//
//  ViewController.m
//  InteractiveCardDemo
//
//  Created by Kitten Yang on 9/28/15.
//  Copyright (c) 2015 Kitten Yang. All rights reserved.
//

#import "ViewController.h"
#import "InteractiveView.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];

    InteractiveView *interactiveView = [[InteractiveView alloc]initWithImage:[UIImage imageNamed:@"pic01"]];
    interactiveView.center = self.view.center;
    interactiveView.bounds = CGRectMake(0, 0, 200, 150);
    interactiveView.gestureView = self.view;
    
    //模糊图层
    UIVisualEffectView *blurView = [[UIVisualEffectView alloc]initWithEffect:[UIBlurEffect effectWithStyle:UIBlurEffectStyleDark]];
    blurView.frame = self.view.bounds;
    [self.view addSubview:blurView];
    interactiveView.dimmingView = blurView;
    
    //interactiveView 的父视图。注意：interactiveView 和 blurView 不能添加到同一个父视图。否则透视效果会使 interactiveView 穿过 blurView
    UIView *backView = [[UIView alloc]initWithFrame:self.view.bounds];
    [self.view addSubview:backView];
    [backView addSubview:interactiveView];
    
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: Swift Version/InteractiveCardDemo-Swift/InteractiveCardDemo-Swift/ViewController.swift
//
//  ViewController.swift
//  InteractiveCardDemo-Swift
//
//  Created by Kitten Yang on 1/6/16.
//  Copyright © 2016 Kitten Yang. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        
        var option = InteractiveOptions()
        option.damping = 0.7
        option.duration = 0.5
        option.scrollDistance = 200.0
        
        let interactiveView = InteractiveView(image: UIImage(named: "pic01"), option: option)
        interactiveView.center = view.center
        interactiveView.bounds = CGRectMake(0, 0, 200, 150)
        interactiveView.gestureView = view

        let blurView = UIVisualEffectView(effect: UIBlurEffect(style: .Dark))
        blurView.frame = view.bounds
        view.addSubview(blurView)
        interactiveView.dimmingView = blurView

        let backView = UIView(frame: view.bounds)
        view.addSubview(backView)
        backView.addSubview(interactiveView)

    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
    }
    
}